mldsa/src/poly.c

```diff
@@ -623,7 +623,8 @@ uint32_t mld_poly_chknorm(const mld_poly *a, int32_t B)
     t |= mld_ct_cmask_neg_i32(B - 1 - mld_ct_abs_i32(a->coeffs[i]));
   }
 
-  return t;
+  /* Return 1, if t = 0xFFFFFFFF */
+  return t & 1;
```

Comment:
If the goal is to not leak which secret is out of bound, this is a problematic change as the compiler can recognize that if you do `t |= ...` repeatedly and in the and merely test `t & 1`, you can abort early once the low-bit is set.